infra/graphs.py

```diff
@@ -42,7 +45,22 @@ def get_baseline_cycles(data, benchmark_name):
   group = [row for row in data if row.get('benchmark', '') == benchmark_name]
   return group_baseline_cycles(group)
 
-def make_plot(profile, upper_x_bound, output):
+def get_cycles(data, benchmark_name, run_method):
+  for row in data:
+    if row.get('benchmark', '') == benchmark_name and row.get('runMethod', '') == run_method:
+      return row.get('cycles')
+  raise KeyError(f"Missing benchmark {benchmark_name} with runMethod {run_method}")
+
+def group_by_benchmark(profile):
+  grouped_by_benchmark = {}
+  for benchmark in profile:
+    benchmark_name = benchmark.get('benchmark', '')
+    if benchmark_name not in grouped_by_benchmark:
+      grouped_by_benchmark[benchmark_name] = []
+    grouped_by_benchmark[benchmark_name].append(benchmark)
+  return [grouped_by_benchmark[benchmark] for benchmark in grouped_by_benchmark]
```

Comment:
why turn it into a list?

Reply:
I've been using a list of rows like `profile.json` as much as possible to keep things uniform